Marlin/src/module/configuration_store.cpp

```diff
@@ -447,36 +447,19 @@ void MarlinSettings::postprocess() {
 
 #if ENABLED(EEPROM_SETTINGS)
 
-  #define WORD_PADDED_EEPROM ENABLED(__STM32F1__, FLASH_EEPROM_EMULATION)
-
-  #if WORD_PADDED_EEPROM && ENABLED(DEBUG_EEPROM_READWRITE)
-    #define UPDATE_TEST_INDEX(VAR) (text_index += sizeof(VAR))
```

Comment:
I think this file should be kept, but there is a typo in current source code... should be "test_index"

Reply:
Yeah that typo was obvious, but there was also others bugs !

- comparing `uint16_t MarlinSettings::datasize() { return sizeof(SettingsData); }` with eeprom_index will always fail at https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/module/configuration_store.cpp#L2039

- in your original code, the odd write should be at `pos + (i*2)` https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/HAL/HAL_STM32F1/persistent_store_flash.cpp#L85

I haven't been able to fix your approach after several hours of debug. So I came down to this new approach.

Reply:
well i didnt touch this file on my branches, and that works..

Reply:
Ahah, I'd bet that.
- Comparing unpadded struct size with padded index cannot work.

- Let's do it manual 
```
...
size = 5
i = 4 
WriteHalfWord(K + (i*2)) = WriteHalfWord(K+8)
i= 5
// loop exit - write odd byte
WriteHalfWord(K + i) = WriteHalfWord(K+5)
...
```

Reply:
The problem changing this file here is you are changing a shared code (with all platforms) just to fix a debug ability on an optional feature of the F1